src/modules/csm/csm.py

```diff
@@ -110,13 +111,13 @@ def build_report(self, blockstamp: ReferenceBlockStamp) -> tuple:
             return ReportData(
                 self.report_contract.get_consensus_version(blockstamp.block_hash),
                 blockstamp.ref_slot,
-                tree_root=prev_root or HexBytes(32),
+                tree_root=prev_root,
                 tree_cid=prev_cid or "",
                 log_cid=log_cid,
                 distributed=0,
             ).as_tuple()
 
-        if prev_cid and prev_root:
+        if prev_cid and prev_root is not ZERO_HASH:
```

Comment:
Didn't quite get why the check for root at all.

Reply:
practically we expect them both to be presented or not presented so i've tried to show it here